applications/FluidDynamicsApplication/custom_constitutive/fluid_constitutive_law.cpp

```diff
@@ -34,6 +34,14 @@ ConstitutiveLaw::Pointer FluidConstitutiveLaw::Clone() const {
     return Kratos::make_shared<FluidConstitutiveLaw>(*this);
 }
 
+FluidAdjointConstitutiveLaw::Pointer FluidConstitutiveLaw::GetAdjointConstitutiveLaw()
+{
+    KRATOS_ERROR << "Calling base FluidConstitutiveLaw::GetAdjointConstitutiveLaw method."
+                 << "Fluid adjoint constitutive law is not defined for " << this->Info()
+                 << std::endl;
+    return Kratos::make_shared<FluidAdjointConstitutiveLaw>(*this);
+}
```

Comment:
Why is this needed?

Reply:
It is not my most favourite also. But the interface I need for CL adjoints does not exists, and it is complicated to make a generic adjoint itnerface for base CL. So I thought of only creating adjoint interface for the fluid only. In this case, I cannot use usual method to assign adjoint CLs because, this interface is not registered in the KratosComponents. So this is the only way I found to get the correct adjoint CL via the user given normal CL.

Reply:
removed :)